packages/connect-core/src/entities/Application.ts

```diff
@@ -101,13 +65,61 @@ export default class Application extends CoreEntity implements ApplicationData {
     this.registryAddress = data.registryAddress
     this.repoAddress = data.repoAddress
     this.version = data.version
+    this.#artifact = data.artifact
+    this.#manifest = data.manifest
+  }
+
+  async _init(): Promise<void> {
+    const {
+      appName,
+      path,
+      functions,
+      deprecatedFunctions,
+      abi,
+    }: AragonArtifact = await resolveMetadata(
+      'artifact.json',
+      this.contentUri!,
+      this.#artifact
+    )
+
+    const {
+      author,
+      description,
+      start_url: htmlUrl,
+      icons,
+      source_url: sourceUrl,
+    }: AragonManifest = await resolveMetadata(
+      'manifest.json',
+      this.contentUri!,
+      this.#manifest
+    )
```

Comment:
We should probably try / catch these two, and emit contextual errors.

Reply:
I decide to handle it in the `resolveMetadata` function. What you think?